feature_engine/encoding/similarity_encoder.py

```diff
@@ -0,0 +1,298 @@
+    'categorical'). You can pass a list of variables to encode. Alternatively, the
+    encoder will find and encode all categorical variables (type 'object' or
+    'categorical').
+
+    With `ignore_format=True` you have the option to encode numerical variables as well.
+    The procedure is identical, you can either enter the list of variables to encode, or
+    the transformer will automatically select all variables.
+
+    The encoder first finds the categories to be encoded for each variable (fit). The
+    encoder then creates one dummy variable per category for each variable
+    (transform).
+
+    Parameters
+    ----------
+    top_categories: int, default=None
+        If None, dummy variables will be created for each unique category of the
+        variable. Alternatively, we can indicate in the number of most frequent
+        categories to encode. In this case, similarity variables will be created
+        only for those popular categories and the rest will be ignored.
+
+    handle_missing : str, default='impute'
+        Action to perform when NaN is seen.
+            'error' - raise an error;
+            'impute' - impute NaN with an empty string;
+            'ignore' - ignore NaN and leave them in resulting columns.
+
+    {variables}
+
+    {ignore_format}
+
+    Attributes
+    ----------
+    encoder_dict_:
+        Dictionary with the categories for which dummy variables will be created.
+
+    {variables_}
+
+    {feature_names_in_}
+
+    {n_features_in_}
+
+    Methods
+    -------
+    fit:
+        Learn the unique categories per variable
+
+    {fit_transform}
+
+    transform:
+        Replace the categorical variables by the binary variables.
+
+    Notes
+    -----
+    This encoder will encode new categories by measuring string similarity between
+    seen and unseen categories.
+
+    No preprocessing is applied, so it's on user to prepare string categorical
+    variables for this transformer.
+
+    The original categorical variables are removed from the returned dataset when we
+    apply the transform() method. In their place, the binary variables are returned.
+
+    See Also
+    --------
+    feature_engine.encoding.OneHotEncoder
+    dirty_cat.SimilarityEncoder
+
+    References
+    ----------
+    .. [1] Cerda P, Varoquaux G, Kégl B. "Similarity encoding for learning with dirty
+       categorical variables". Machine Learning, Springer Verlag, 2018.
+    .. [2] Cerda P, Varoquaux G. "Encoding high-cardinality string categorical
+       variables". IEEE Transactions on Knowledge & Data Engineering, 2020.
+    """
+
+    def __init__(
+        self,
+        top_categories: Union[None, int] = None,
+        handle_missing: str = 'impute',
+        variables: Union[None, int, str, List[Union[str, int]]] = None,
+        ignore_format: bool = False,
+    ):
+        if top_categories and not isinstance(top_categories, int):
```

Comment:
For the error messages could we add f"Got {param_value} instead."

We are rolling that out for all new classes, and slowly updating the old ones.